DCAT-2088: Add instructions for generating API references (#167)
AdobeDocs/commerce-services: README.md

   yarn dev
   ```

> **Tip:** When you encounter unexpected failures or delays, try to clean up the gatsby and yarn caches with the command: `yarn clean && yarn cache clean` for troubleshooting.

Author: ```suggestion
**Tip:** If you encounter unexpected failures or delays, try to clean up the gatsby and yarn caches with the command: `yarn clean && yarn cache clean` for troubleshooting.
```

- [Arranging content structure](https://github.com/adobe/aio-theme#content-structure)
- [Linking to pages](https://github.com/adobe/aio-theme#links)
- [Using assets](https://github.com/adobe/aio-theme#assets)
- [Configuring global navigation](https://github.com/adobe/aio-theme#global-navigation)
- [Configuring side navigation](https://github.com/adobe/aio-theme#side-navigation)
- [Using content blocks](https://github.com/adobe/aio-theme#jsx-blocks)
- [Writing enhanced Markdown](https://github.com/adobe/aio-theme#writing-enhanced-markdown)
- [Deploying the site](https://github.com/adobe/aio-theme#deploy-to-azure-storage-static-websites) _(Adobe employees only)_
- [Render OpenAPI spec](https://github.com/adobe/aio-theme?tab=readme-ov-file#openapi)
- [Use RedoclyAPIBlock to render OpenAPI spec](https://github.com/adobe/aio-theme?tab=readme-ov-file#redoclyapiblock)

If you have questions, open an issue and ask us. We look forward to hearing from you!

## GraphQL API reference generator

The GraphQL API reference is generated using an open-source tool [SpectaQL](https://github.com/anvilco/spectaql). The data required for the generator is located at the `spectaql` directory:

> - `adobe-theme`: [custom theme](https://github.com/anvilco/spectaql/blob/main/examples/themes/README.md).

Reviewer: We haven't been using a custom theme in this project yet.

- `config.yml`: [configuration file](https://github.com/anvilco/spectaql#yaml-options).

> - `schema.json`: the GraphQL schema extracted from the Adobe Commerce (B2B) instance.

Reviewer: We use a URL to the resources that return schemas instead of storing a schema file in this project.

> The resulting GraphQL API reference lives in the `static/graphql-api/` directory.

Reviewer: This project contains two references.

> It is embedded into the `/graphql/reference` page using the `frameSrc` feature on the DevSite.

Reviewer: Wrong path to the page. There are two pages with references.

> To rebuild the GraphQL API references after any updates, use yarn to run the following build scripts as needed:

Reviewer: ```suggestion
To rebuild the GraphQL API references after any updates, use yarn locally to run the following build scripts as needed:
```

> ```shell

Reviewer: Add text descriptions for each yarn script name instead of the actual script code.
Example: 
- `build:admin-api` - build the Admin API reference.

"build:admin-api": "spectaql --target-file index.html --config spectaql/config-admin.yml",
"dev:admin-api": "spectaql --development-mode-live --config spectaql/config-admin.yml",
"build:storefront-api": "spectaql --target-file index.html --config spectaql/config-storefront.yml",
"dev:storefront-api": "spectaql --development-mode-live --config spectaql/config-storefront.yml",
"build:graphql": "yarn run build:admin-api && yarn run build:storefront-api"
```

For example, to rebuild the Catalog management and rules API, run the command:

```shell
yarn build:admin-api
```

The dev scripts run spectaql in development mode for a live preview of the updated output.

### How to get the schema

The Spectaql configuration files for the CCDM GraphQL API references use the following endpoints to retrieve the schemas and generate the API reference:

- Catalog management and rules API: https://commerce-admin-router-qa.corp.ethos501-stage-va6.ethos.adobe.net/graphql
- Storefront API: https://catalog-service-qa.adobe.io/graphql

> If either of these endpoints change, be sure to update the Spectaql config file and rebuild the API reference guides.

Reviewer: It's not clear why you should update the config file.

### Update the API References

If a schema changes, rebuild and test the API reference locally.  Then, submit a PR with updates against the `ccdm-early-access` branch.

> After the PR is merged, someone from the documentation team will publish the changes to the documentation server.

Reviewer: ```suggestion
After the PR is merged, the documentation team will publish the changes to the documentation website.
```

> ### Resources

Reviewer: Duplication of the Resources clause.

- `config-storefront.yml`: [configuration file to generate the CCDM Storefront API Reference](spectaql/config-admin.yml).

These configuration files include the endpoint for each API service.

When you build the API reference, the build script uses live introspection to retrieve the GraphQL schemas and generate the API references. The configuration file also provides the introductory text for the Welcome topic, the API reference title, and other settings used when generating the references.

The resulting GraphQL API references are output to the `static/graphql-api/` directory.

- `static/graphql/admin-api/index.html`
- `static/graphql/storefront-api/index.html`

The references are embedded in the API Reference page using the `frameSrc` feature supported by the Adobe I/O theme.

- [Catalog management and rules API Reference](src/pages/composable-catalog/admin/api-reference.md) `frameSrc: /graphql-api/admin-api/index.html`

- [Storefront API reference](src/pages/composable-catalog/storefront-services/api-reference.md) `frameSrc: /graphql-api/storefront-api/index.html`

> `

Author: ```suggestion
```

- [Storefront API reference](src/pages/composable-catalog/storefront-services/api-reference.md) `frameSrc: /graphql-api/storefront-api/index.html`

> `

Reviewer: ```suggestion

```

Reviewer: Redundant symbol

> - [Storefront API reference](https://github.com/AdobeDocs/commerce-services/edit/ccdm-early-access/src/pages/composable-catalog/storefront-services/api-reference.md) `frameSrc: /graphql-api/storefront-api/index.html`

Reviewer: ```suggestion
```
Duplication

If a schema changes, rebuild and test the API reference locally.  Then, submit a PR with updates against the `ccdm-early-access` branch. After the PR is merged, someone from the documentation team will publish the changes to the documentation server.

For local builds, ensure that your environment has the following installed:

> - Node.js that matches the version set in the [.nvmrc](https://github.com/AdobeDocs/commerce-services/blob/main/.nvmrc) npm configuration file.

Reviewer: ```suggestion
- Node.js that matches the version set in the [.nvmrc](https://github.com/AdobeDocs/commerce-services/blob/main/.nvmrc) ([nvm](https://github.com/nvm-sh/nvm) configuration file).
```
